orte/mca/plm/rsh/plm_rsh_module.c

```diff
@@ -1518,40 +1518,38 @@ static int setup_shell(orte_plm_rsh_shell_t *rshell,
                        char *nodename, int *argc, char ***argv)
 {
     orte_plm_rsh_shell_t remote_shell, local_shell;
-    char *param;
     int rc;
-
+    FILE *fp;
+    char path[256];
+    bool skip;
+    
     /* What is our local shell? */
     local_shell = ORTE_PLM_RSH_SHELL_UNKNOWN;
 
-#if OPAL_ENABLE_GETPWUID
-    {
-        struct passwd *p;
 
-        p = getpwuid(getuid());
-        if( NULL == p ) {
-            /* This user is unknown to the system. Therefore, there is no reason we
-             * spawn whatsoever in his name. Give up with a HUGE error message.
-             */
-            orte_show_help( "help-plm-rsh.txt", "unknown-user", true, (int)getuid() );
-            return ORTE_ERR_FATAL;
-        }
-        param = p->pw_shell;
-        local_shell = find_shell(p->pw_shell);
+    /* Open the command for reading. */
+    fp = popen("ps -p $$ | awk '{print $4}'", "r");
```

Comment:
Do we know that "ps -p $$" will work in all environments?  E.g., does it work in the various BSDs?  Does it work on Solaris?